master/static/migrations/20230710120538_migrate-checkpointv1.tx.up.sql

```diff
@@ -0,0 +1,110 @@
+-- Endstate of checkpoint views / tables is
+-- raw_checkpoints is left unmodified.
+-- checkpoints_v2 has checkpoint_v2s and unarchived checkpoint_v1s.
+-- checkpoints_view has all unarchived checkpoints pulling from checkpoints_v2.
+-- proto_checkpoints_view is the proto version of checkpoints_view.
+--
+-- Note we just leave checkpoints_v1 data so we can reverse this migration.
+INSERT INTO public.checkpoints_v2 (
+    uuid,
+    task_id,
+    allocation_id,
+    report_time,
+    state,
+    resources,
+    metadata,
+    size
+)
+SELECT
+    -- https://stackoverflow.com/questions/12505158/generating-a-uuid-in-postgres-for-insert-statement
+    COALESCE(c.uuid, uuid_in(overlay(overlay(md5(random()::text || ':' || random()::text) placing '4' from 13) placing to_hex(floor(random()*(11-8+1) + 8)::int)::text from 17)::cstring)),
+    t.task_id,
+    CASE -- TODO is this behaviour okay? Or should we backfill and insert into allocations.
+        WHEN a.allocation_id IS NULL THEN NULL
+        ELSE a.allocation_id
+    END,
+    COALESCE(c.end_time, t.end_time, NOW()),
+    c.state,
+    c.resources,
+    c.metadata,
```

Comment:
can you verify the structure of metadata didnt change from checkpoint v1 -> v2. (cc @ashtonG , checkpoint metadata is potentially a good candidate for 'bad use of JSON column')

Reply:
Good call

Old checkpoint metadata was more user submitted labels, the checkpoint view actually handles this but I missed carrying over this handling. Updated.